### PR TITLE
[Model] Support Qwen3ForSequenceClassification for Qwen3-Embed Model

### PR DESCRIPTION
**Pull request by nysa-liu**
## Motivation

Support the newly released **Qwen3-embed** series models by extending the classification capabilities of Qwen3.

To this end, I implemented `Qwen3ForSequenceClassification` based on the existing Qwen3 script and `Qwen2ForSequenceClassification`. This enables users to perform sequence classification tasks with Qwen3-embed models.

## Modifications

* Created a new file: `qwen3_classification.py` to define `Qwen3ForSequenceClassification`.
* Updated `docs/supported_models/embedding_models.md` to include relevant documentation for Qwen3-embed classification support.
* Functionality has been tested on both NVIDIA 40-series and A100 GPUs.

## Checklist

* [x] Format your code according to the [[Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit)](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
* [x] Add unit tests as outlined in the [[Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci)](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
* [x] Update documentation / docstrings / example tutorials as needed, according to [[Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci)](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
* [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [[Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html)](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [[Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html)](https://docs.sglang.ai/references/accuracy_evaluation.html).
* [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
* [x] Please feel free to join our Slack channel at [https://slack.sglang.ai/](https://slack.sglang.ai) to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/models/qwen3_classification.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        prefix: str = "",
+    ) -> None:
+        super().__init__()
+        self.config = config
+        self.quant_config = quant_config
+        self.model = Qwen3Model(
+            config, quant_config=quant_config, prefix=add_prefix("model", prefix)
+        )
+        self.score = nn.Linear(config.hidden_size, config.num_labels)
+        # Use normalize=True for qwen3 embedding based on official implementation
```
Comment (yizhang2077) [inline python/sglang/srt/models/qwen3_classification.py:42]: could you give a reference here?

File: python/sglang/srt/models/qwen3_classification.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
+        # Filter out lm_head weights of Qwen3ForCausalLM
+        filtered_weights = [
+            (name, w) for name, w in weights if not name.startswith("lm_head")
+        ]
+        return Qwen3ForCausalLM.load_weights(self, filtered_weights)
+
+
+EntryClass = [
+    Qwen3ForSequenceClassification,
```
Comment (yizhang2077) [inline python/sglang/srt/models/qwen3_classification.py:77]: I think we could add `Qwen3ForSequenceClassification` into `is_generation_model` function logic

File: python/sglang/srt/models/qwen3_classification.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+]
```
Comment (yizhang2077) [inline python/sglang/srt/models/qwen3_classification.py]: could we add ut in `test_embedding_models.py` as `Qwen2ForSequenceClassification` does?

File: docs/supported_models/embedding_models.md [hunk 4]
```diff
@@ -42,13 +42,14 @@ response = requests.post(url + "/v1/embeddings", json=payload).json()
...
+| **Qwen3 (QwenEmbeddingModel)**             | `Qwen/Qwen3-Embedding-4B`              | N/A            | Qwen3-based text embedding model (4B), compatible with QwenEmbeddingModel interface; designed for accurate and efficient semantic representation.  |
+| **GME (MultimodalEmbedModel)**             | `Alibaba-NLP/gme-Qwen2-VL-2B-Instruct` | `gme-qwen2-vl` | Multimodal embedding model (2B) based on Qwen2‑VL, encoding image + text into a unified vector space for cross‑modal retrieval.                    |
+| **CLIP (CLIPEmbeddingModel)**              | `openai/clip-vit-large-patch14-336`    | N/A            | OpenAI’s CLIP model (ViT‑L/14) for embedding images (and text) into a joint latent space; widely used for image similarity search.                 |
+| **BGE (BgeEmbeddingModel)**                | `BAAI/bge-large-en-v1.5`               | N/A            | Currently only support `attention-backend`   `triton` and `torch_native`. BAAI's BGE embedding models optimized for retrieval and reranking tasks. |
```
Comment (yizhang2077) [inline docs/supported_models/embedding_models.md]: could you provide more details about how to use `Qwen3ForSequenceClassification` as  https://github.com/vllm-project/vllm/pull/19260 does?

File: python/sglang/srt/models/qwen3_classification.py (hunk 2 above)
Comment (nysa-liu) [reply python/sglang/srt/models/qwen3_classification.py:77]: **For adding Qwen3ForSequenceClassification to is_generation_model logic:**

I've made the following changes:

1. **Added to is_generation_model function**: Added `"Qwen3ForSequenceClassification"` to the list of non-generative model architectures in `python/sglang/srt/configs/model_config.py`.

**Comment by nysa-liu** [comment]
Thanks for the feedback. I've updated the implementation accordingly. Please help take another look 🙏 @yizhang2077

**Comment by yizhang2077** [comment]
Sorry for late reply. Instead of changing other model's descriptions, I think we only need add *Qwen3-Embedding* part?

File: python/sglang/srt/models/qwen3_classification.py (hunk 1 above)
Comment (yizhang2077) [reply python/sglang/srt/models/qwen3_classification.py:42]: It seems this logic depends on whether we use `embedding model` or `rerank model`. This logic is right for `embedding model`, it looks like in `rerank model` it uses softmax in pooler, could we change pooler `normalize` or `softmax` according to model architectures?
